File: app/src/main/java/io/github/gusandrianos/foxforreddit/data/models/PollData.java
package io.github.gusandrianos.foxforreddit.data.models;

import java.util.List;
import com.google.gson.annotations.SerializedName;

public class PollData{

    @SerializedName("voting_end_timestamp")
    private long votingEndTimestamp;

    @SerializedName("total_stake_amount")
    private Object totalStakeAmount;

    @SerializedName("is_prediction")
    private boolean isPrediction;

    @SerializedName("user_selection")
    private Object userSelection;

    @SerializedName("options")
    private List<OptionsItem> options;

    @SerializedName("resolved_option_id")
    private Object resolvedOptionId;

    @SerializedName("user_won_amount")
    private Object userWonAmount;

    @SerializedName("total_vote_count")
    private int totalVoteCount;

    public long getVotingEndTimestamp(){
        return votingEndTimestamp;
    }

    public Object getTotalStakeAmount(){
        return totalStakeAmount;
    }

    public boolean isIsPrediction(){
        return isPrediction;
    }

    public Object getUserSelection(){
        return userSelection;
    }

    public List<OptionsItem> getOptions(){
        return options;
    }

    public Object getResolvedOptionId(){
        return resolvedOptionId;
    }

    public Object getUserWonAmount(){
        return userWonAmount;
    }

    public int getTotalVoteCount(){
        return totalVoteCount;
    }
}

File: app/src/main/java/io/github/gusandrianos/foxforreddit/data/models/PollData.kt
package io.github.gusandrianos.foxforreddit.data.models

import com.google.gson.annotations.SerializedName

class PollData {
    @SerializedName("voting_end_timestamp")
    val votingEndTimestamp: Long = 0

    @SerializedName("total_stake_amount")
    val totalStakeAmount: Any? = null

    @SerializedName("is_prediction")
    val isIsPrediction = false

    @SerializedName("user_selection")
    val userSelection: Any? = null

    @SerializedName("options")
    val options: List<OptionsItem>? = null

    @SerializedName("resolved_option_id")
    val resolvedOptionId: Any? = null

    @SerializedName("user_won_amount")
    val userWonAmount: Any? = null

    @SerializedName("total_vote_count")
    val totalVoteCount = 0
}